Game/RainMeadow.GameplayHooks.cs

```diff
@@ -155,21 +158,19 @@ bool WeaponIsDangerous(Weapon weapon)
 
         private bool Weapon_HitThisObject(On.Weapon.orig_HitThisObject orig, Weapon self, PhysicalObject obj)
         {
-            if (!self.IsLocal())
+            if (OnlineManager.lobby != null)
             {
-                return false;
-            }
+                if (ModManager.MSC && obj is Player pl && pl.slugOnBack?.slugcat != null && pl.slugOnBack.slugcat == self.thrownBy)
+                {
+                    return false;
+                }
 
-            if (obj is Creature c && c.FriendlyFireSafetyCandidate(self.thrownBy) && WeaponIsDangerous(self))
-            {
-                return false;
-            }
+                if (obj is Creature c && WeaponIsDangerous(self))
+                {
+                    return !c.FriendlyFireSafetyCandidate(self.thrownBy);
```

Comment:
Story code was falling through to default hook strangely enough, but never worked. If I manually set story.friendlyFire above this line, it worked fine. So I figure we put this at the bottom of the online logic and return its inverted bool.

Reply:
Looks good.

Check the origin to preserve mod compat
```suggestion
                    return orig(self, obj) && !c.FriendlyFireSafetyCandidate(self.thrownBy);
```